src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -105,10 +110,20 @@ class UtxoState(override val persistentProver: PersistentBatchAVLProver[Digest32
 
         log.debug(s"Trying to apply full block with header ${fb.header.encodedId} at height $height")
 
+        implicit val es = constants.settings
         val inRoot = rootHash
-
-        val stateTry = stateContext.appendFullBlock(fb).flatMap { newStateContext =>
-          applyTransactions(fb.blockTransactions.txs, fb.header.stateRoot, newStateContext).map { _: Unit =>
+        val blockData = BlockMetricData(fb.id, fb.height,
```

Comment:
why blockdata is allocated always (while metrics will be in 99% cases switched off)

dangerous complication here as well.